network-api/networkapi/wagtailpages/wagtail_hooks.py

```diff
@@ -100,3 +101,14 @@ def before_delete_page(request, page):
     if isinstance(page, ProductPage) and page.votes:
         # Delete the vote from ProductPages
         page.votes.delete()
+
+
+@hooks.register('after_delete_page')
+@hooks.register('after_publish_page')
+@hooks.register('after_unpublish_page')
+def manage_pni_cache(request, page):
+    if isinstance(page, ProductPage):
```

Comment:
We'll want to also invalidate the cache if the BuyersGuidePage gets updated because that'll invalidate which products to show (e.g. the cutoff date changes, etc)